app/models/vote.rb

```diff
@@ -2,6 +2,8 @@ class Vote < ApplicationRecord
 
   belongs_to :votable, polymorphic: true
 
+  validates :user_id, :votable_id, :votable_type, presence: true
+
```

Comment:
I think you'll also want an application-level `uniqueness` validation here for the combination of `user_id`, `votable_id`, and `votable_type`.  Check out the bit on the `:scope` option in the [validation Rails docs](http://guides.rubyonrails.org/active_record_validations.html#uniqueness).